src/ast/codegen_llvm.cpp

```diff
@@ -1884,7 +1894,14 @@ void CodegenLLVM::visit(AssignVarStatement &assignment)
   }
   else
   {
-    b_.CreateStore(expr_, variables_[var.ident]);
+    Value *val = expr_;
+    if (assignment.expr->type.IsArrayTy())
+    {
+      val = b_.CreatePtrToInt(expr_, b_.getInt64Ty());
+      // Since only the pointer is copied, we need to extend lifetime of RHS
+      scoped_del.disarm();
```

Comment:
Are you sure this is what you want? Lifetime extension is usually done with:
```
expr_deleter_ = scoped_del.disarm();
```

Reply:
That will only extend lifetime to the end of the current statement but I need to extend it to the end of the probe (since other statements may want to read from that pointer). I haven't found any other way to do that, so this seemed like the simplest solution.

Anyway, I plan to improve work with structs and arrays (to avoid unnecessary memcpys), so I'll probably come with a cleaner solution, then.